src/map/channel.c

```diff
@@ -278,12 +279,21 @@ void channel_send(struct channel_data *chan, struct map_session_data *sd, const
 		clif->messagecolor_self(sd->fd, COLOR_RED, msg_sd(sd,1455));
 		return;
 	} else if (sd) {
+		int i;
+
 		safesnprintf(message, 150, "[ #%s ] %s : %s", chan->name, sd->status.name, msg);
 		clif->channel_msg(chan,sd,message);
 		if (chan->type == HCS_TYPE_IRC)
 			ircbot->relay(sd->status.name,msg);
 		if (chan->msg_delay != 0)
 			sd->hchsysch_tick = timer->gettick();
+
+		for (i = 0; i < MAX_EVENTQUEUE; i++) {
+			if (chan->handlers[i][0] != '\0') {
+				script->set_var(sd, "@channelmes$", aStrdup(msg));
```

Comment:
This seems a memory leak: `script->set_var()` doesn't take ownership of the string you pass (it takes a const pointer), but creates its own copy internally. The call sequence is:

1. `script_set_var()` calls `script->set_reg()`, passing the const pointer as is (`value`) and passes no reg_db (`ref == NULL`)
2. `script_set_reg()` calls `pc->setregstr()` if the variable if of type string and `ref` is NULL. Once again, it passes the const pointer as is (casts from `const void *` to `const char *`.
3. `pc_setregstr()` checks that the string isn't empty (zero length). If so, it duplicates it through `aStrdup()`.

You can simply pass `msg`, without duplicating it.